egs/wsj/s5/steps/dict/select_prons_bayesian.py

```diff
@@ -0,0 +1,415 @@
+          ", sorted by their counts in acoustic training data, "
+          ,file=args.ref_lexicon_edits_handle)
+    print("# 1st Col: source of the candidate pron: G(2P) / P(hone-decoding) / R(eference)."
+          ,file=args.ref_lexicon_edits_handle)
+    print("# 2nd Col: soft counts from lattice-alignment (not augmented by prior-counts)."
+          ,file=args.ref_lexicon_edits_handle)
+    print("# 3rd Col: accepted or not in the learned lexicon (Y/N).", file=args.ref_lexicon_edits_handle)
+    print("# 4th Col: the pronunciation cadidate.", file=args.ref_lexicon_edits_handle)
+    
+    # words which are to be printed into the edits file.
+    words_to_edit = [] 
+    for word in learned_lexicon:
+        count = counts.get(word, 0)
+        flags = ['0' for i in range(3)] # "flags" contains three binary indicators, 
+        # indicating where this word's pronunciations come from.
+        for pron in learned_lexicon[word]:
+            if word in phone_decoding_lexicon and pron in phone_decoding_lexicon[word]:
+                flags[0] = '1'
+                src[(word, pron)] = 'P'
+            if word in ref_lexicon and pron in ref_lexicon[word]:
+                flags[1] = '1'
+                src[(word, pron)] = 'R'
+            if word in g2p_lexicon and pron in g2p_lexicon[word]:
+                flags[2] = '1'
+                src[(word, pron)] = 'G'
+        if word in ref_lexicon:
+            all_ref_prons_accepted = True
+            for pron in ref_lexicon[word]:
+                if pron not in learned_lexicon[word]:
+                    all_ref_prons_accepted = False
+                    break
+            if not all_ref_prons_accepted or flags[0] == '1' or flags[2] == '1':
+                words_to_edit.append((word, counts[word]))
+            if count > thr:
+                words[0][flags[0] + flags[1] + flags[2]].add(word)
+            else:
+                words[1][flags[0] + flags[1] + flags[2]].add(word)
+        else:
+            if count > thr: 
+                words[2][flags[0] + flags[2]].add(word)
+            else:
+                words[3][flags[0] + flags[2]].add(word)
+
+    words_to_edit_sorted = sorted(words_to_edit, key=lambda entry: entry[1], reverse=True)
+    for word, count in words_to_edit_sorted:
+        print("------------",word, "%2.1f" % count, "--------------", file=args.ref_lexicon_edits_handle)
+        for pron in learned_lexicon[word]:
+            print(src[(word, pron)], ' | Y | ', "%2.1f | " % stats.get((word, pron), 0), pron, 
+                  file=args.ref_lexicon_edits_handle)
+        for pron in ref_lexicon[word]:
+            if pron not in learned_lexicon[word]:
+                print('R', ' | N | ', "%2.1f | " % stats.get((word, pron), 0), pron, 
+                      file=args.ref_lexicon_edits_handle)
+    print("Here are the words whose reference pron candidates were all declined", words[0]['100'])
```

Comment:
Try to avoid extremely long lines in the source code.
Also, this would be much more readable if you set some variables with informative names instead of using those expressions directly, and used .format() instead of string concatenation.